packages/react/src/KeybindingHint/key-names.ts

```diff
@@ -0,0 +1,113 @@
+/** Converts the first character of the string to upper case and the remaining to lower case. */
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+const capitalize = ([first, ...rest]: string) => (first?.toUpperCase() ?? '') + rest.join('').toLowerCase()
+
+// In the below records, we don't intend to cover every single possible key - only those that
+// would be realistically used in shortcuts. For example, the Pause/Break key is not necessary
+// because it is not found on many keyboards.
+
+/**
+ * Short-form iconic versions of keys. These should be intuitive (not archaic) and match icons on keyboards.
+ */
+export const condensedKeyName = (key: string, isMacOS: boolean) =>
+  ({
+    alt: isMacOS ? '⌥' : 'Alt', // the alt key _is_ the option key on MacOS - in the browser there is no "option" key
+    control: '⌃',
+    shift: '⇧',
+    meta: isMacOS ? '⌘' : 'Win',
+    mod: isMacOS ? '⌘' : '⌃',
+    pageup: 'PgUp',
+    pagedown: 'PgDn',
+    arrowup: '↑',
+    arrowdown: '↓',
+    arrowleft: '←',
+    arrowright: '→',
+    plus: '+', // needed to allow +-separated chords
+    backspace: '⌫',
+    delete: 'Del',
+    space: '␣', // needed to allow space-separated sequences
+    tab: '⇥',
+    enter: '⏎',
+    escape: 'Esc',
+    function: 'Fn',
+    capslock: 'CapsLock',
+    insert: 'Ins',
+    printscreen: 'PrtScn',
+  })[key] ?? capitalize(key)
+
+/**
+ * Specific key displays for 'full' format. We still do show some icons (ie punctuation)
+ * because that's more intuitive, but for the rest of keys we show the standard key name.
+ */
+export const fullKeyName = (key: string, isMacOS: boolean) =>
+  ({
+    alt: isMacOS ? 'Option' : 'Alt',
+    mod: isMacOS ? 'Command' : 'Control',
+    '+': 'Plus',
+    pageup: 'Page Up',
+    pagedown: 'Page Down',
+    arrowup: 'Up Arrow',
+    arrowdown: 'Down Arrow',
+    arrowleft: 'Left Arrow',
+    arrowright: 'Right Arrow',
+    capslock: 'Caps Lock',
+    printscreen: 'Print Screen',
+  })[key] ?? capitalize(key)
+
+/**
+ * Accessible key names intended to be read by a screen reader. This prevents screen
+ * readers from expressing punctuation in speech, ie, reading a long pause instead of the
+ * word "period".
+ */
+export const accessibleKeyName = (key: string, isMacOS: boolean) =>
```

Comment:
How confident are we with the accessible names? Have we confirmed with the accessibility team? Just wanted to make sure there is some ground work for this.

Reply:
I'm pretty sure we discussed this with the accessibility team in Slack way back when we first wrote this component, but that was almost two years ago so unfortunately I don't have any record of it. I do feel pretty confident in this based on my own screen reader testing, but I am not the target audience so I can't say that with absolute certainty.

Reply:
Thank you! I'll share this with the team to see if we need to cross check this again with the Accessibility team. Thanks for your patience 🙏🏻